tests/integration_tests/test_analysis_results_file.py

```diff
@@ -0,0 +1,61 @@
+"""
+Test reading and writing analysis results.
+
+"""
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from unittest import mock
+
+from fab.dep_tree import AnalysedFile
+from fab.steps.analyse import Analyse
+from fab.util import HashedFile
+
+
+def test_load_analysis_results():
+    # tests:
+    #   new file
+    #   changed file
+    #   previously analysed file, unchanged
+    #   file no longer there
+
+    # source folder before change
+    previous_file_hashes = {
+        Path('change.f90'): 111,
+        Path('no_change.f90'): 222,
+        Path('remove.f90'): 333,
+    }
+
+    # source folder after change
+    latest_file_hashes = {
+        Path('change.f90'): 123,
+        Path('no_change.f90'): 222,
+        Path('new.f90'): 444,
+    }
+
+    previous_results = [
+        AnalysedFile(fpath=path, file_hash=file_hash) for path, file_hash in previous_file_hashes.items()]
+
+    with TemporaryDirectory() as tmpdir:
+        analyser = Analyse(root_symbol=None)
+
+        # simulate the effect of calling run, in which the superclass sets up the _config attribute (is this too ugly?)
+        analyser._config = mock.Mock(workspace=Path(tmpdir))
+
+        # run 1
+        # create the previous analysis file
+        with analyser._new_analysis_file(unchanged=previous_results):
```

Comment:
I think test classes can reasonably be considered "friend" classes. However, we expect to split up the analysis file so this will probably be resolved naturally then.